### PR TITLE
[MU4] Added logger

### PR DESCRIPTION
**Pull request by igorkorsukov**
Example output:
```
13:22:44.978 | ERROR | main_thread | AccountController | init: Error while init
13:22:44.980 | DEBUG | main_thread | ShortcutsRegister | expandStandartKeys: added 3 shortcut, because they are alternative shortcuts for the given standard keys
13:22:44.980 | WARN  | main_thread | ExtensionsConfiguration | fileList: 401The file does not exist
13:22:44.980 | WARN  | main_thread | ExtensionsConfiguration | fileList: 401The file does not exist
13:22:44.982 | WARN  | main_thread | WorkspaceManager | setupCurrentWorkspace: filed get workspace: , will use Basic
13:22:45.182 | INFO  | main_thread | AudioEngine | init: success inited audio engine
13:22:45.185 | DEBUG | main_thread | FluidSynth | init: synth inited
```

Currently, log files stored in the `data/logs`
for example `HOME/.local/share/MuseScore/MuseScore4Development/logs`

### CODE REVIEW

File: framework/global/globalmodule.cpp [hunk 1]
```diff
@@ -40,25 +46,37 @@ std::string GlobalModule::moduleName() const
...
+    //! --- Setup logger ---
+    using namespace haw::logger;
+    Logger* logger = Logger::instance();
+    logger->clearDests();
+
+    //! Console
+    logger->addDest(new ConsoleLogDest(LogLayout("${time} | ${type|5} | ${thread} | ${tag|10} | ${message}")));
+
+    //! File, this creates a file named "data/logs/MuseScore_yyMMdd.log"
+    std::string logsPath = io::path(s_globalConf->dataPath() + "/logs").toStdString();
```
Comment (RomanPudashkin) [inline framework/global/globalmodule.cpp:64]: Maybe add a function to configuration:

io::path logsDirPath() { return dataPath() + "/logs" }

?

**Review by RomanPudashkin — APPROVED** [review]

File: framework/global/CMakeLists.txt [hunk 2]
```diff
@@ -21,17 +21,24 @@ set(MODULE global)
 
 include(${CMAKE_CURRENT_LIST_DIR}/modularity/modularity.cmake)
 include(${CMAKE_CURRENT_LIST_DIR}/async/async.cmake)
+include(${PROJECT_SOURCE_DIR}/thirdparty/haw_logger/logger/logger.cmake)
```
Comment (Eism) [inline framework/global/CMakeLists.txt:24]: Can you explain to me what does 'haw' mean?

File: framework/global/globalmodule.cpp (hunk 1 above)
Comment (igorkorsukov) [reply framework/global/globalmodule.cpp]: added `logsPath` method to configuration

File: framework/global/CMakeLists.txt (hunk 2 above)
Comment (igorkorsukov) [reply framework/global/CMakeLists.txt:24]: It's just a name that doesn't mean anything :)
(hawthorn)

**Review by Eism — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
Plans to add translatability?

**Comment by igorkorsukov** [comment]
> Plans to add translatability?

No, the log should not be translatable, it is for developers, not for users.
It will be difficult for us, as developers, to analyze the log in other languages, for example, Chinese.

**Comment by Jojo-Schmitz** [comment]
I see, makes sense.

**Pull request merged**